src/coreComponents/constitutive/fluid/ReactiveMultiFluid.hpp

```diff
@@ -0,0 +1,224 @@
+                                   std::move( phaseEnthalpy ),
+                                   std::move( phaseInternalEnergy ),
+                                   std::move( phaseCompFraction ),
+                                   std::move( totalDensity )  ),
+    m_numPrimarySpecies( numPrimarySpecies ),
+    m_equilibriumReactions( equilibriumReactions.createKernelWrapper() ),
+    m_kineticReactions( kineticReactions.createKernelWrapper() ),
+    m_primarySpeciesConcentration( primarySpeciesConcentration ),
+    m_secondarySpeciesConcentration( secondarySpeciesConcentration ),
+    m_primarySpeciesTotalConcentration( primarySpeciesTotalConcentration ),
+    m_kineticReactionRates( kineticReactionRates )
+    {}
+    
+  protected:
+
+    friend class ReactiveMultiFluid;
+    /// Reaction related terms
+    integer m_numPrimarySpecies;
+
+    chemicalReactions::EquilibriumReactions::KernelWrapper m_equilibriumReactions;
+
+    chemicalReactions::KineticReactions::KernelWrapper m_kineticReactions;
+
+    arrayView2d< real64 >  m_primarySpeciesConcentration;
+
+    arrayView2d< real64 >  m_secondarySpeciesConcentration;
+
+    arrayView2d< real64 >  m_primarySpeciesTotalConcentration;
+
+    arrayView2d< real64 >  m_kineticReactionRates;
+  };
+
+  struct viewKeyStruct : ConstitutiveBase::viewKeyStruct
+  {};
+
+protected:
+
+  virtual void postProcessInput() override;
+
+  void createChemicalReactions();
+
+  virtual void resizeFields( localIndex const size, localIndex const numPts ) override;
+
+  /// Reaction related terms
+  integer m_numPrimarySpecies;
+
+  integer m_numSecondarySpecies;
+
+  integer m_numKineticReactions;
+
+  std::unique_ptr< chemicalReactions::EquilibriumReactions > m_equilibriumReactions;
+
+  std::unique_ptr< chemicalReactions::KineticReactions > m_kineticReactions;
+
+  array2d< real64 >  m_primarySpeciesConcentration;
+
+  array2d< real64 >  m_secondarySpeciesConcentration;
+
+  array2d< real64 >  m_primarySpeciesTotalConcentration;
+
+  array2d< real64 >  m_kineticReactionRates;
+};
+
+GEOSX_HOST_DEVICE
+inline void
+ReactiveMultiFluid::KernelWrapper::
+  computeChemistry( real64 const pressure,
+                    real64 const temperature,
+                    arraySlice1d< real64 const, compflow::USD_COMP - 1 > const & composition,
+                    arraySlice1d< real64, compflow::USD_COMP - 1 > const & primarySpeciesConcentration,
+                    arraySlice1d< real64, compflow::USD_COMP - 1 > const & secondarySpeciesConcentration,
+                    arraySlice1d< real64, compflow::USD_COMP - 1 > const & primarySpeciesTotalConcentration,
+                    arraySlice1d< real64, compflow::USD_COMP - 1 > const & kineticReactionRates ) const
+{
+  // I am assuming that the primary variable is the concentration of the primary species.
+  for( int i=0; i < m_numPrimarySpecies; i++ )
+  {
+    primarySpeciesTotalConcentration[i] = composition[i];
+  }
```

Comment:
Just a small question: is the `composition` coming in the right unit here (when coming from the compositional solver)? 

I am just asking because I saw in the `ReactiveMultiFluid` driver:
```
feedComposition="{3.76e-1, 3.76e-1, 3.87e-2, 3.21e-2, 1.89, 1.65e-2, 1.09 }"
```
with compositions above 1.

Reply:
yeah, I think these are `mol/L`.  unit conversion is kind of a pain which is why I have not taken care of it yet.